File: app/models/charges.scala
package models

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.regex.Pattern;
import java.util.concurrent.TimeUnit;

import org.junit._;

import org.junit.Assert._
import org.hamcrest.CoreMatchers._

import org.openqa.selenium._;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.net.InetAddress; 
import java.net.UnknownHostException; 
 
import org.slf4j.Logger; 
import org.slf4j.LoggerFactory;

import com.datastax.driver.core.Cluster; 
import com.datastax.driver.core.Session; 
import com.datastax.driver.core.querybuilder.QueryBuilder; 



object Charge {
  
  val driver: WebDriver = new FirefoxDriver();
  driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);

  
  
  def launch(url: String, prefix: String, firstname: String, lastname: String, email: String, address1: String, city: String, state: String, zip: String) {
   
    
    driver.get(url);
    
    
  }
  

}

File: app/models/charges.java
package models;

import java.util.List;
import java.util.concurrent.TimeUnit;

import org.junit.*;

import static org.junit.Assert.*;

import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.phantomjs.PhantomJSDriver;
import org.openqa.selenium.phantomjs.PhantomJSDriverService;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.WebDriverWait;



public class Charges {
  private boolean acceptNextAlert = true;
  private StringBuffer verificationErrors = new StringBuffer();
  
  
  public WebDriver selectDriver(Boolean ghostProtocol) {
	  if (ghostProtocol) {
		  DesiredCapabilities caps = new DesiredCapabilities();
	      caps.setJavascriptEnabled(true);
	      caps.setCapability("takesScreenshot", true);
	      caps.setCapability(PhantomJSDriverService.PHANTOMJS_EXECUTABLE_PATH_PROPERTY,"/usr/local/bin/phantomjs");
	      return new PhantomJSDriver(caps);
	  } else {
		  return new FirefoxDriver();
	  } 
  }
  
  public String launch(Boolean ghostProtocol, String url, String prefix, String firstname, String lastname, String email, String address1, String city, String state, String zip) throws Exception {

	  WebDriver driver = selectDriver(ghostProtocol);
      driver.get(url);      
      new Select(driver.findElement(By.id("title"))).selectByVisibleText(prefix);
      driver.findElement(By.id("fname")).clear();
      driver.findElement(By.id("fname")).sendKeys(firstname);    
      driver.findElement(By.id("lname")).clear();
      driver.findElement(By.id("lname")).sendKeys(lastname);
      driver.findElement(By.id("street1")).clear();
      driver.findElement(By.id("email")).sendKeys(email);
      driver.findElement(By.id("street1")).clear();
      driver.findElement(By.id("street1")).sendKeys(address1);
      driver.findElement(By.id("city")).clear();
      driver.findElement(By.id("city")).sendKeys(city);
      new Select(driver.findElement(By.id("state"))).selectByVisibleText(state);    
      driver.findElement(By.id("zip")).clear();
      driver.findElement(By.id("zip")).sendKeys(zip); 
      driver.findElement(By.id("send")).click(); 

      Thread.sleep(5000);
      
      // check what the page title is if it is thanks we can quit if it is asking for more info then give it more info!
      
      String endPageTitle = "Page title is: " + driver.getTitle();
      
      //Close the browser
      driver.quit();	
      
      return endPageTitle;

  }
  
  
  
}
